src/Validator.php

```diff
@@ -50,13 +51,13 @@ public function __construct(
         /**
          * @var bool|callable|null
          */
-        $defaultSkipOnEmpty = null,
+        bool|callable|null $defaultSkipOnEmpty = null,
     ) {
         $this->defaultSkipOnEmptyCallback = SkipOnEmptyNormalizer::normalize($defaultSkipOnEmpty);
     }
 
     /**
-     * @param DataSetInterface|mixed|RulesProviderInterface $data
+     * @param LimitInterface|mixed|RulesProviderInterface $data
```

Comment:
```suggestion
     * @param mixed $data
```

Reply:
Oops, this is a leftover of an accidental wrong refactoring (all occurrences of `DataSetInterface` was replaced by `LimitInterface`. 

So before the changes it was:

```php
@param DataSetInterface|mixed|RulesProviderInterface $data
```

I restored it to previous version for now (no changes at all). `mixed` in this context indeed need some improvements. But I'm against applying this suggestion:

- It just matches type hint.
- Auto completion in IDE is worse.
- According Interface needs adjustments too.
- 
Maybe postpone it for a future PR. By the way, after reverting a new Psalm error appeared and needs to be fixed.

Reply:
```
ERROR: MoreSpecificImplementedParamType - src/Validator.php:62:36 - Argument 1 of Yiisoft\Validator\Validator::validate has the more specific type 'array<array-key, mixed>|object', expecting 'Yiisoft\Validator\DataSetInterface|Yiisoft\Validator\RulesProviderInterface|mixed' as defined by Yiisoft\Validator\ValidatorInterface::validate (see https://psalm.dev/140)
    public function validate(mixed $data, iterable|object|string|null $rules = null): Result
```